packages-exp/analytics-exp/src/factory.ts

```diff
@@ -218,18 +215,29 @@ export function factory(
 
     globalInitDone = true;
   }
+
+  const analyticsInstance: AnalyticsService = new AnalyticsService(app);
+  return analyticsInstance;
+}
+
+/**
+ * Starts async initializations for this app.
+ */
+export function _initializeAnalyticsForApp(
+  app: FirebaseApp,
+  installations: _FirebaseInstallationsInternal,
+  options?: AnalyticsOptions
+): void {
   // Async but non-blocking.
   // This map reflects the completion state of all promises for each appId.
-  initializationPromisesMap[appId] = initializeAnalytics(
+  // Factory will have thrown if options has no appId.
+  initializationPromisesMap[app.options.appId!] = _initializeAnalytics(
```

Comment:
After giving it some thought, I realize we can't do initialization outside the factory, otherwise FCM, which depends on Analytics, might try to use an uninitialized Analytics service, as shown in the below example:

```
import { initializeApp } from 'firebase/app';
import { getMessaging, onMessage } from 'firebase/messaging';
import { getAnalytics } from 'firebase/analytics';

const app = initializeApp({...});

/**
 * Messaging calls `const analytics = await messaging.firebaseDependencies.analyticsProvider.get();` 
 * which will create and return an uninitialized Analytics service.
 * And it will probably throw when calling `analytics.logEvent(...)`
 */
const msg = getMessaging(app);
onMessage(msg, () => { 
    // process messages
});

// getAnalytics async after a timeout
setTimeout(() => getAnalytics(), 1000);
```

We should let analytics initialize with the default options in this situation.
To support this,  some changes to the component framework need to be made, e.g. allow it to take an option and pass it to the factory.

I need to think more about it and let's chat offline.  This PR will need to be put on hold for now, Sorry!

Reply:
https://github.com/firebase/firebase-js-sdk/pull/4595 is merged. We can use `Provider.initialize()` to initialize Auth in the factory function.